src/ProvidedTypes.fs

```diff
@@ -13729,7 +13729,21 @@ namespace ProviderImplementation.ProvidedTypes
         let isAddress s = (s = ExpectedStackState.Address)
         let rec emitLambda(callSiteIlg: ILGenerator, v: Var, body: Expr, freeVars: seq<Var>, lambdaLocals: Dictionary<_, ILLocalBuilder>, parameters) =
             let lambda: ILTypeBuilder = assemblyMainModule.DefineType(UNone, genUniqueTypeName(), TypeAttributes.Class)
-            let baseType = convTypeToTgt (typedefof<FSharpFunc<_, _>>.MakeGenericType(v.Type, body.Type))
+
+            let fsharpFuncType = convTypeToTgt (typedefof<FSharpFunc<_, _>>)
+            let voidType = convTypeToTgt typeof<System.Void>
+            let rec lambdaType (t : Type) = 
+                if t.IsGenericType then 
+                    let args = t.GetGenericArguments()
+                    let gdef = t.GetGenericTypeDefinition()
+                    if args.Length = 2 && gdef.FullName = fsharpFuncType.FullName && args.[1] = voidType then 

```

Comment:
you can use [`FSharp.Reflection.FSharpType.IsFunction`](https://msdn.microsoft.com/visualfsharpdocs/conceptual/fsharptype.isfunction-method-%5bfsharp%5d) to test if a type is a F# function, and [`FSharp.Reflection.FSharpType.GetFunctionElements`](https://msdn.microsoft.com/visualfsharpdocs/conceptual/fsharptype.getfunctionelements-method-%5bfsharp%5d) to get the types of the domain and range of the function, checking those for the approproate matching types.

Reply:
The type has already been converted to the target context

Reply:
Admittedly I just assumed it wouldn't work and didn't try it. But I ran a quick test to confirm and I get `Message: System.NotSupportedException `